com.unity.render-pipelines.universal/Documentation~/writing-shaders-urp-basic-unlit-structure.md

```diff
@@ -132,17 +131,46 @@ Tags { "RenderType" = "Opaque" "RenderPipeline" = "UniversalPipeline" }
 
+You can create multiple SubShaders with different `RenderPipeline` tags if you wish to execute the same shader in different render pipelines.
+
 For more information on SubShader Tags, see [ShaderLab: SubShader Tags](https://docs.unity3d.com/Manual/SL-SubShaderTags.html).
 
 ### Pass block
 
 In this example, there is one Pass block that contains the HLSL program code. For more information on Pass blocks, see [ShaderLab: Pass](https://docs.unity3d.com/Manual/SL-Pass.html).
 
+A Pass block can optionally contain a Pass Tags block. [ShaderLab: PassTags](https://docs.unity3d.com/Manual/SL-PassTags.html).
+
+When writing passes for URP you need to define the `LightMode` that tells the pipeline which pass to use when executing different parts of the render pipeline.
+
+Possible values for `LightMode` tag are:
+
+*UniversalForward*: Renders objects geometry and evaluate all light contributions. Used in forward rendering mode.
+
+*UniversalGBuffer*: Renders objects geometry without evaluating any light contribution. Used in deferred rendering mode.
+
+*UniversalForwardOnly*: Same as *UniversalForward*, however used for both forward and deferred rendering modes. This is useful when you want to render objects in forward while the pipeline is in deferred mode because they contain additional data that doens't fit the GBuffer, f.ex, clear coat normals. If you want a shader to render in both forward and deferred, declare both 
+
+*UniversalForward* and *UniversalGBuffer* passes. If you want a material to render in forward regarless of the rendering mode you should declare only a *UniversalForwardOnly* pass.
+
+*Universal2D*: Renders objects and evaluate 2D lights. Used in 2D renderer.
+
+*ShadowCaster*: Renders object depth from the perspective of lights into the shadowmap or a depth texture.
+
+*DepthOnly*: Renders only depth information from the perspective of the camera into a depth texture.
+
+*Meta*: Used by the Lightmapper to bake lightmaps and probes. This pass is stripped from shaders when building a standalone player.
+
+*SRPDefaultUnlit*: Used to draw an additional pass when rendering objects. Use can use this to draw an object outline. Used in both forward and deferred rendering mode. When a pass doesn't declare a LightMode tag, it behaves as if *SRPDefaultUnlit* was declared.
+
+> **NOTE**: The following LightMode tags are not supported in URP: Always, ForwardAdd, PrepassBase, PrepassFinal, Vertex, VertexLMRGBM, VertexLM.
+
+
 ### HLSLPROGRAM block
 
 This block contains the HLSL program code.
 
-> **NOTE**: URP shaders support only the HLSL language.
+> **NOTE**: URP supports shaders written with CGPROGRAM/ENDCGPROGRAM, however when using those, Unity will automatically include the built-in shader library and if you also include SRP shader library you might end up with conflicting shader libraries. It's also not possible to have a shader that's SRP Batcher compatible when using CGPROGRAM.
```

Comment:
Is it correct to write it so:
"HLSL language is the preferred language for URP shaders."
Separate note:
"URP supports the CG language. If you add the CGPROGRAM/ENDCGPROGRAM block in a shader, Unity includes shaders from the Built-in Render Pipeline library automatically. If you include shaders from the SRP shader library, some SRP shader macros and functions might conflict with the Buit-in Render Pipeline shader functions. Shaders with the CGPROGRAM block are not SRP Batcher compatible."